crates/driver/src/domain/competition/mod.rs

```diff
@@ -224,6 +224,9 @@ impl Competition {
                         tracing::warn!(block = block.number, ?err, "solution reverts on new block");
                         *score_ref = None;
                         *self.settlement.lock().unwrap() = None;
+                        if let Some(id) = settlement.notify_id() {
```

Comment:
Here we have IF because we decided to not send the notification for merged solutions. While this is okkish now since external solvers are sending a single solution anyway, it may become problematic in the future with combinatorial auctions...